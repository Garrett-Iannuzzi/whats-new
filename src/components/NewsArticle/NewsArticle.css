.article-individual {
  border: solid 1px darkgrey;
  font-size: .9em;
  margin: .5em;
  padding: 1em;
  width: 48%;
}

.article-individual:hover {
  border: solid 1px indianred;
  background-color: whitesmoke;
}

.art-img {
  border: solid 2px darkgray;
  width: 70%;
}

.article-title {
  font-size: 1em;
  padding: .7em;
}

.article-description {
  font-size: .8em;
  padding: 15px;
}

.see-article-btn {
  border-radius: 5px;
  font-weight: bold;
  height: 2em;
  margin: 1em;
  width: 40%;
}

.see-article-btn:hover {
  cursor: pointer;
  color: white;
  background-color: indianred;
}