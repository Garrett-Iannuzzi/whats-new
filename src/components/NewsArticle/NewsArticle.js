import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ title, img, description, url }) => {
  return (
    <article>
      <h3>{title}</h3>
      <image>{img}</image>
      <p>{description}</p>
      <button>See Article</button>
    </article>
  )
}

export default NewsArticle;