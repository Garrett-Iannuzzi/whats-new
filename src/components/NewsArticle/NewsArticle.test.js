import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';


describe('News Article', () => {

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      title='News'
      img='//www.images.com'
      description='No news is good news'
      />);
      expect(wrapper).toMatchSnapshot();
  });
});
