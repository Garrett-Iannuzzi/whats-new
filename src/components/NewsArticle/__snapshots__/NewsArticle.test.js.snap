// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Article should match the snapshot with all data passed in correctly 1`] = `
<article
  className="article-individual"
>
  <h3
    className="article-title"
  >
    News
  </h3>
  <img
    alt="article img"
    className="art-img"
    src="//www.images.com"
  />
  <p
    className="article-description"
  >
    No news is good news
  </p>
  <a>
    <button
      className="see-article-btn"
    >
      See Article
    </button>
  </a>
</article>
`;
